### ISSUE TITLE
Duplicate parameter with name body and location body

### ISSUE THREAD

**Issue by dgiebert** [marshmallow-code/apispec#546]
Created a rather simple REST API with the following setup:

Model
```
class Question(db.Model):
    __tablename__ = "initial_questions"
    id = Column(Integer, primary_key=True, autoincrement=True)
    text = Column(Text, nullable=False)
```

Schema
```
class QuestionSchema(ma.SQLAlchemyAutoSchema):
    class Meta:
        model = Question

    @post_load
    def load(self, data, **kwargs):
        return Question(**data)
```

Controller
```
questions_blueprint = Blueprint("questions", __name__, url_prefix="/questions")

@questions_blueprint.route("/", methods=["POST", "GET"])
@jwt_required()
def controller():
    question_controller = QuestionController(request)
    return question_controller(request)

class QuestionController(MethodResource):
    @use_kwargs(QuestionSchema)
    @marshal_with(QuestionSchema(many=True))
    def get(self, **kwargs):
        try:
            return Question.query.filter_by(domain=question_domain).all()
        except StatementError as e:
            return make_response(str(e.orig), 400)
        except Exception as e:
            return make_response("", 500)
```
Config + Register:
```
app.config.update({
    'APISPEC_SPEC': APISpec(
        title='Co-Rona',
        version='v1',
        openapi_version="3.0.2",
        plugins=[MarshmallowPlugin()],
    ),
    'APISPEC_SWAGGER_URL': '/swagger/',
})

apispec = FlaskApiSpec(app)
apispec.register(QuestionController, blueprint="questions", endpoint="controller")
```

This leads to the following error:
```
apispec.exceptions.DuplicateParameterError: Duplicate parameter with name name and location body
```

After a "bit" of digging the reason is the property parsing that defaults to naming all elements in 'body' to 'body'
See here: https://github.com/marshmallow-code/apispec/blob/dev/src/apispec/ext/marshmallow/openapi.py#L232

For me changing this to the passed name attribute fixes the problem.

**Comment by lafrech** [comment]
Thanks for the report. I guess you're using flask-apispec.

Can you provide a reproducible example? A single piece of code with imports and all.

**Comment by dgiebert** [comment]
### Sample Code
```
from apispec import APISpec
from apispec.ext.marshmallow import MarshmallowPlugin
from flask import Flask, Blueprint, make_response
from flask_apispec import MethodResource, FlaskApiSpec, use_kwargs
from flask_marshmallow import Marshmallow
from flask_sqlalchemy import SQLAlchemy

db = SQLAlchemy()
ma = Marshmallow()

# Blueprints
class Model(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(50))

class Schema(ma.ModelSchema):
    class Meta:
        model = Model

class Controller(MethodResource):
    @use_kwargs(Schema())
    def post(self, domain, **kwargs):
        return make_response("Test", 200)

blueprint = Blueprint("test", __name__, url_prefix="/test")
blueprint.add_url_rule('/', view_func=Controller.as_view('test'))

app = Flask(__name__)
app.register_blueprint(blueprint)

app.config.update({
    'APISPEC_SPEC': APISpec(
        title='Test',
        version='v1',
        openapi_version="3.0",
        plugins=[MarshmallowPlugin()],
    ),
    'APISPEC_SWAGGER_URL': '/swagger/',
})

apispec = FlaskApiSpec(app)
apispec.register(Controller, blueprint="test", endpoint="test")

app.run()
```

### requirements.txt
```
Flask==1.0.3
Flask-Script==2.0.6
Flask-SQLAlchemy==2.4.0
flask-marshmallow==0.11.0
flask-apispec==0.8.7
marshmallow-sqlalchemy==0.22.3
```

Yes it is using the flask-apispec

**Comment by Bangertm** [comment]
TLDR: your code will work if you change ```openapi_version="3.0"``` to ```openapi_version="2.0"```

The problem here is that in APISpec `Component.path` is getting called with `operations` equal to 
```python
{'post': {'parameters': [
   {'in': 'body', 'name': 'body', 'required': False, 'schema': {'type': 'object', 'properties': {'name': {'type': 'string', 'nullable': True, 'maxLength': 50}}}},
   {'in': 'body', 'name': 'body', 'required': False, 'schema': {'type': 'object', 'properties': {'id': {'type': 'integer', 'format': 'int32'}}}}
], 'responses': {}}}
```
This is invalid because both parameters have the same `name` parameter of body.

The source of this problem is in flask-apispec. It looks like the `Converter.get_parameters` is not taking the OpenAPI version into account and is assuming that the OpenAPI version is 2. In this case it is calling `schema2parameters` with a body parameter, which is something that is invalid for OpenAPI 3. To handle this properly, flask-apispec should include logic for to properly format a schema in a request body - either in parameters for OpenAPI 2 or in requestBody for OpenAPI 3.

**Comment by lafrech** [comment]
Closing this.

Please report the issue here to flask-apispec.
